docs/platform-issues.rst

```diff
@@ -0,0 +1,227 @@
+Platform-specific tips and tricks
+==================================
+
+Linux
+------
+
+Possible conflicts with Bash
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Depending on how your installation of Bash is configured, Xonsh may have trouble
+loading certain shell modules. If you see errors similar to this
+when launching Xonsh:
+
+.. code-block:: console
+
+    bash: module: line 1: syntax error: unexpected end of file
+    bash: error importing function definition for 'BASH_FUNC_module'
+    bash: scl: line 1: syntax error: unexpected end of file
+    bash: error importing function definition for 'BASH_FUNC_scl'
+    bash: module: line 1: syntax error: unexpected end of file
+    bash: error importing function definition for 'BASH_FUNC_module'
+    bash: scl: line 1: syntax error: unexpected end of file
+    bash: error importing function definition for 'BASH_FUNC_scl'
+
+...You can correct the problem by unsetting the modules, by adding the following
+lines to your ``~/.bashrc file``:
+
+.. code-block:: console
+
+    unset module
+    unset scl
+
+
+Default Ubuntu .bashrc breaks Foreign Shell Functions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Xonsh supports importing functions from foreign shells using the
+`ForeignShellFunctionAlias` class, which calls functions as if they were
+aliases. This is implemented by executing a command that sources the file
+containing the function definition and then immediately calls the function with
+any necessary arguments.
+
+The default user `~/.bashrc` file in Ubuntu 15.10 has the following snippet at
+the top, which causes the script to exit immediately if not run interactively.
+
+.. code-block:: bash
+
+    # If not running interactively, don't do anything
+    case $- in
+        *i*) ;;
+          *) return;;
+    esac
+
+This means that any function you have added to the file after this point will be
+registered as a xonsh alias but will fail on execution if the 
+shell is not running interactively. Previous versions of
+Ubuntu have a different test for interactivity at the top of the file that
+yields the same problem.
+
+
+MacOS, OSX
+----------
+
+readline
+^^^^^^^^
+
+[ed note: This recommendation seems to be `out of date <https://pypi.org/project/gnureadline/>`_. 
```

Comment:
Can you rephrase to make it seem less like the docs are contradicting themselves. Maybe soften the lang below?